airflow/migrations/versions/0e2a74e0fc9f_add_time_zone_awareness.py

```diff
@@ -0,0 +1,205 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Add time zone awareness
+
+Revision ID: 0e2a74e0fc9f
+Revises: d2ae31099d61
+Create Date: 2017-11-10 22:22:31.326152
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '0e2a74e0fc9f'
+down_revision = 'd2ae31099d61'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+from sqlalchemy.dialects import mysql
+import sqlalchemy as sa
+
+
+def upgrade():
```

Comment:
Given that run_id = id_prefix + run_date, I wonder if ideally run_id should also be updated to reflect the new time in utc. Given that run_id is never parsed again, the only potential confusion is that there's a bit of semantic inconsistency with future generated run_ids

Reply:
Checked the schema and this field has a unique constraint, so it may potentially cause collision with new run_ids created after the migration.

Reply:
Run_id will be in UTC by definition as execution_date is by definition time zone aware

Reply:
ah you are right, I was running Airflow locally with system clock (PST) before upgrading db.

Reply:
Can you be a bit more clear what happened? Although you were running with PST the upgrade should happen with UTC. ie `2017-01-01 01:00:00` should just become `2017-01-01 01:00:00+00:00`. For run_ids this just means it becomes something like `scheduled__2017-01-01 01:00:00+00:00`.

Reply:
ah yes, that is the case for any new run_ids that get generated, but not for past ones since the migration scripts only updates 'execution_date`, `start_date`, and `end_date` in the table.

Reply:
Ah, ok. I think we *might* need to update run_ids (I rather not) in the migration script as well as they are sometimes used in look ups. I need to verify if it will be an issue if we don't. I think it might be an issue for backfills, but not sure.

Reply:
^^ seems not to be required. We never use the full run_id.